### PR TITLE
Save and restore placement modes when calling vnic_cfg

### PR DESCRIPTION
**Pull request by stephen-hurd**
It seems that vnic_cfg changes the placement mode on some (all?) firmware versions.

### CODE REVIEW

File: drivers/net/bnxt/bnxt_hwrm.c [hunk 1]
```diff
@@ -897,12 +904,67 @@ int bnxt_hwrm_vnic_alloc(struct bnxt *bp, struct bnxt_vnic_info *vnic)
 	return rc;
 }
 
+static int bnxt_hwrm_vnic_plcmodes_qcfg(struct bnxt *bp, struct bnxt_vnic_info *vnic, struct bnxt_plcmodes_cfg *pmode)
+{
+	int rc = 0;
+	struct hwrm_vnic_plcmodes_qcfg_input req = {.req_type = 0 };
+	struct hwrm_vnic_plcmodes_qcfg_output *resp = bp->hwrm_cmd_resp_addr;
+
+	HWRM_PREP(req, VNIC_CFG, -1, resp);
```
Comment (ajitkhaparde) [inline drivers/net/bnxt/bnxt_hwrm.c]: Should this be:
HWRM_PREP(req, VNIC_PLCMODES_QCFG, -1, resp);

File: drivers/net/bnxt/bnxt_hwrm.c [hunk 2]
```diff
@@ -897,12 +904,67 @@ int bnxt_hwrm_vnic_alloc(struct bnxt *bp, struct bnxt_vnic_info *vnic)
...
+	return rc;
+}
+
+static int bnxt_hwrm_vnic_plcmodes_cfg(struct bnxt *bp, struct bnxt_vnic_info *vnic, struct bnxt_plcmodes_cfg *pmode)
+{
+	int rc = 0;
+	struct hwrm_vnic_plcmodes_cfg_input req = {.req_type = 0 };
+	struct hwrm_vnic_plcmodes_cfg_output *resp = bp->hwrm_cmd_resp_addr;
+
+	HWRM_PREP(req, VNIC_CFG, -1, resp);
```
Comment (ajitkhaparde) [inline drivers/net/bnxt/bnxt_hwrm.c]: Should this be:
HWRM_PREP(req, VNIC_PLCMODES_CFG, -1, resp);

**Comment by ajitkhaparde** [comment]
So the plc mode settings that the VF driver was setting up, is lost without this change?

**Comment by stephen-hurd** [comment]
Correct, vnic_cfg sets a hard-coded placement mode.

**Review by ajitkhaparde — APPROVED** [review]

**Pull request merged**